www/js/diary/infinite_scroll_list.js

```diff
@@ -319,45 +377,95 @@ angular.module('emission.main.diary.infscrolllist',['ui-leaflet',
     $ionicScrollDelegate.scrollBottom();
     ClientStats.addReading(ClientStats.getStatKeys().LABEL_TAB_SWITCH, {"source": prev, "dest": $scope.getActiveFilters()});
   }
 
-  $scope.recomputeDisplayTimelineEntries = function() {
+  $scope.updateLoadedRange = (oldestTrip, latestTrip) => {
+    if (oldestTrip && oldestTrip.start_ts <= $scope.infScrollControl.pipelineRange.start_ts) {
+      Logger.log("Oldest trip in batch starts at "+ moment(oldestTrip.start_ts)
+          +" pipeline starts at "+moment($scope.infScrollControl.pipelineRange.start_ts)
+          +". Reached the start of pipeline, nothing earlier to load");
+      $scope.infScrollControl.oldestLoadedTs = oldestTrip.start_ts;
+      $scope.infScrollControl.reachedPipelineStart = true;
+    }
+    if (latestTrip && latestTrip.end_ts >= $scope.infScrollControl.pipelineRange.end_ts) {
+      Logger.log("Latest trip in batch ends at "+ moment(latestTrip.end_ts)
+          +" pipeline ends at "+moment($scope.infScrollControl.pipelineRange.end_ts)
+          +". Reached the end of pipeline, nothing later to load");
+      $scope.infScrollControl.reachedPipelineEnd = true;
+      $scope.infScrollControl.latestLoadedTs = latestTrip.end_ts;
+    }
+  }
+
+  $scope.updateTimeRange = (dir, oldestTrip, latestTrip) => {
+    $scope.updateLoadedRange(oldestTrip, latestTrip);
+    if (!dir || dir == 'past') {
+      if (oldestTrip) {
+          if (oldestTrip.start_ts > $scope.infScrollControl.pipelineRange.start_ts) {
+              // Since this was reversed, the first entry is the most recent
+              $scope.infScrollControl.oldestLoadedTs =
+                  oldestTrip.end_ts - 1;
+              Logger.log("new oldest loaded time = "+$scope.infScrollControl.oldestLoadedTs);
+          }
+      } else {
+          Logger.log("current batch of size 0 but haven't reached pipeline start, going on");
+          $scope.infScrollControl.oldestLoadedTs = $scope.infScrollControl.oldestLoadedTs - ONE_WEEK;
+      }
+    }
+    if (!dir || dir == 'future') {
+      if (latestTrip) {
+        if (latestTrip.end_ts < $scope.infScrollControl.pipelineRange.end_ts) {
+            $scope.infScrollControl.latestLoadedTs =
+                latestTrip.end_ts + 1;
+            Logger.log("new latest loaded time = "+$scope.infScrollControl.latestLoadedTs);
+        }
+      } else {
+          Logger.log("current batch of size 0 but haven't reached pipeline end, going on");
+          $scope.infScrollControl.latestLoadedTs = $scope.infScrollControl.latestLoadedTs + ONE_WEEK;
+      }
+    }
+    $scope.infScrollControl.selRangeText = moment($scope.infScrollControl.oldestLoadedTs*1000).format("L")
+                                        + "\n" + moment($scope.infScrollControl.latestLoadedTs*1000).format("L");
+    const rangeMiddle = ($scope.infScrollControl.oldestLoadedTs + $scope.infScrollControl.latestLoadedTs)/2;
+    $scope.infScrollControl.selectedDay = moment(rangeMiddle*1000).toDate();
```

Comment:
This is what ensures that we select the middle of the selected range without having to scroll a lot?!

Reply:
Yes- as a concrete example if you have loaded trips in the range of July 22 - September 9, the datepicker is set on August 13. So if you click the datepicker, August pops up at first.